test/e2e/operator_test.go

```diff
@@ -101,6 +122,72 @@ func TestDefaultClusterIngressExists(t *testing.T) {
 	}
 }
 
+func TestClusterIngressControllerCreateDelete(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ci := &ingressv1alpha1.ClusterIngress{}
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: "default"}, ci); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to get default IngressController: %v", err)
+	}
+
+	dnsConfig := &configv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
+		t.Fatalf("failed to get DNS 'cluster': %v", err)
+	}
+
+	ing := newIngressController(ingressControllerName, ns, ingressControllerName+"."+dnsConfig.Spec.BaseDomain)
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		if err := cl.Create(context.TODO(), ing); err != nil {
+			if errors.IsAlreadyExists(err) {
+				return true, nil
+			}
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("failed to create IngressController %s/%s: %v", ing.Namespace, ing.Name, err)
```

Comment:
Should be `Fatalf`.

Reply:
I'll update. I thought we wanted to Errorf on create so we can still proceed to deleting.